core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -1746,6 +1746,29 @@ object Chunk extends ChunkFactory with ChunkPlatformSpecific {
       BitChunk(bytes.drop(toDrop), min, max)
     }
 
+    private def bitwise(that: BitChunk, f: (Boolean, Boolean) => Boolean): BitChunk = {
```

Comment:
In order to be efficient, `f` should be `(Byte, Byte) => Byte`, so it can apply a bitwise operation to the whole bytes. This will complicate the bitwise operations because it's not just `l & r` unless the bytes are aligned (i.e. unless both min bit indices are evenly divisible by 8).

You could try in this PR just to get the parent operation merged in, leaving optimized overrides in BitChunk for subsequent tickets.

Reply:
Hey there, I am working on the Byte => Byte => Byte version!